source/extensions/filters/http/ext_proc/ext_proc.cc

```diff
@@ -100,6 +100,20 @@ FilterHeadersStatus Filter::decodeHeaders(RequestHeaderMap& headers, bool end_st
 
 Http::FilterDataStatus Filter::onData(ProcessorState& state, ProcessingMode::BodySendMode body_mode,
                                       Buffer::Instance& data, bool end_stream) {
+  if (state.callbackState() == ProcessorState::CallbackState::Headers) {
+    ENVOY_LOG(trace, "Header processing still in progress -- holding body data");
+    if (end_stream) {
+      // Buffer the data and we'll send the callback when the response callback returns.
```

Comment:
`send the data`?

maybe `when we finish header processing`?

Also just by reading this code I'm wondering if we know for sure that we're going to be sending the data or if there are scenarios where we might? If so should the language be less direct (e.g. `we may send the data once we get a response`)?

Reply:
I re-wrote the comments -- I hope that they are more clear now.